lib/retriable.rb

```diff
@@ -6,14 +6,22 @@
 module Retriable
   module_function
 
-  def self.configure
+  def configure
     yield(config)
   end
 
   def config
     @config ||= Config.new
   end
 
+  def retriable_with_context(context_key, options = {}, &block)
+    if !config.context.key?(context_key)
+      raise ArgumentError, "#{context_key} not found in Retriable.config.context"
+    end
+
+    retriable(config.context[context_key].merge(options), &block) if block
+  end
+
   def retriable(opts = {})
```

Comment:
if you're going to declare the &block in the `with_retriable` args, might as well add it here too.  might help generations to come read the code slightly.